### PR TITLE
Graql Quickcheck Property Tests

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

File: grakn-test/test-integration/src/test/java/ai/grakn/generator/VarPatternAdmins.groovy [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import ai.grakn.graql.Var;
+import ai.grakn.graql.admin.VarPatternAdmin;
+import ai.grakn.graql.admin.VarProperty;
+import ai.grakn.graql.internal.pattern.Patterns;
+
+/**
+ * @author Felix Chapman
+ */
+public class VarPatternAdmins extends RecursiveGenerator<VarPatternAdmin> {
```
Comment (aelred) [inline grakn-test/test-integration/src/test/java/ai/grakn/generator/VarPatternAdmins.groovy]: Somehow this is a groovy file! Whoops...

File: grakn-test/test-integration/src/test/java/ai/grakn/test/property/PatternPropertyTests.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        VarPatternAdmin varPattern1 = Patterns.varPattern(var1, properties);
+        VarPatternAdmin varPattern2 = Patterns.varPattern(var2, properties);
+        assertNotEquals(varPattern1, varPattern2);
+    }
+
+    @Property
+    public void theConjunctionOfTwoPatterns_ShouldBeContainedInTheResultingPattern(Pattern pattern1, Pattern pattern2){
+        Set<VarPattern> union = new HashSet<>();
+        union.addAll(pattern1.admin().varPatterns());
+        union.addAll(pattern2.admin().varPatterns());
```
Comment (aelred) [inline grakn-test/test-integration/src/test/java/ai/grakn/test/property/PatternPropertyTests.java]: Guava has a nice thing:
```
Set<VarPattern> union = Sets.union(pattern1.admin().varPatterns(), pattern2.admin().varPatterns());
```

It's up to you which to use though.

File: grakn-test/test-integration/src/test/java/ai/grakn/generator/VarPatternAdmins.groovy (hunk 1 above)
Comment (fppt) [reply grakn-test/test-integration/src/test/java/ai/grakn/generator/VarPatternAdmins.groovy]: Ooops :)

File: grakn-test/test-integration/src/test/java/ai/grakn/test/property/PatternPropertyTests.java (hunk 2 above)
Comment (fppt) [reply grakn-test/test-integration/src/test/java/ai/grakn/test/property/PatternPropertyTests.java]: Why did I not use that . . . Oh well. I will change it now

**Comment by fppt** [comment]
!rtg - ai.grakn.client.BatchExecutorClientTest.whenEngineRESTFailsWhileLoadingWithRetryTrue_LoaderRetriesAndWaits

**Pull request merged**